Can't include boost .hpp for pure build

With this manifest:
```
version = 1

[install]
gcc.pkg-path = "gcc"
boost.pkg-path = "boost"


[build.test_boost]
command = """
    g++ -o test_boost test_boost.cpp
    mkdir -p $out/bin
    cp test_boost $out/bin/test_boost
"""
runtime-packages = []
sandbox = "pure"
```

And this `test_boost.cpp`:
```
#include <iostream>
#include <string>
#include <boost/lexical_cast.hpp>
int main() {
    try {
        std::string str = "123";
        int num = boost::lexical_cast<int>(str);
        std::cout << num << std::endl;
    }
    catch (const boost::bad_lexical_cast& e) {
        std::cerr << "Lexical cast error: " << e.what() << std::endl;
    }
    return 0;
}
```

`flox build` fails with:
```
test_boost> 00:00:00.326224 test_boost.cpp:3:10: fatal error: boost/lexical_cast.hpp: No such file or directory
test_boost> 00:00:00.326224     3 | #include <boost/lexical_cast.hpp>
test_boost> 00:00:00.326224       |          ^~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/assets/environment-interpreter/activate/activate
+++ b/assets/environment-interpreter/activate/activate
@@ -167,8 +167,8 @@ while true; do
       ;;
     -m | --mode)
       shift
-      if [ -z "${1:-}" ] || ! { [ "$1" == "run" ] || [ "$1" == "dev" ] || [ "$1" == "build" ] || [ "$1" == "build-sandbox" ]; }; then
-        echo "Option --mode requires 'dev', 'run', 'build', or 'build-sandbox' as an argument." >&2
+      if [ -z "${1:-}" ] || ! { [ "$1" == "run" ] || [ "$1" == "dev" ] || [ "$1" == "build" ]; }; then
+        echo "Option --mode requires 'dev', 'run', or 'build' as an argument." >&2
         echo "$USAGE" >&2
         exit 1
       fi
@@ -258,7 +258,7 @@ case "$_flox_shell" in
     ;;
 esac
 
-if [ "$_FLOX_ENV_ACTIVATION_MODE" = "build" ] || [ "$_FLOX_ENV_ACTIVATION_MODE" = "build-sandbox" ]; then
+if [ "$_FLOX_ENV_ACTIVATION_MODE" = "build" ]; then
   if [ $# -eq 0 ]; then
     echo "build mode is only supported for running a command" >&2
     exit 1

--- a/cli/flox-rust-sdk/src/providers/build.rs
+++ b/cli/flox-rust-sdk/src/providers/build.rs
@@ -1978,12 +1978,10 @@ mod tests {
         boost_include_only(false);
     }
 
-    // TODO: currently failing since build-sandbox mode doesn't run etc-profiles
-    // for the outer activation
-    // #[test]
-    // fn boost_include_only_sandbox_pure() {
-    //     boost_include_only(true);
-    // }
+    #[test]
+    fn boost_include_only_sandbox_pure() {
+        boost_include_only(true);
+    }
 
     #[test]
     fn cleans_up_data_sandbox() {

--- a/package-builder/Makefile
+++ b/package-builder/Makefile
@@ -1,6 +1,6 @@
 .DEFAULT_GOAL := all
 
-ALL = flox-build.mk build-manifest.nix nef/default.nix \
+ALL = env-filter flox-build.mk build-manifest.nix nef/default.nix \
   $(addprefix nef/lib/,default.nix dirToAttrs.nix extendAttrSet.nix mkOverlay.nix reflect.nix)
 
 OS := $(shell uname -s)
@@ -19,6 +19,10 @@ libsandbox.so: sandbox.c closure.o
 
 libsandbox.dylib: sandbox.c closure.o
 	$(CC) -pthread -dynamiclib $^ -o $@
+
+env-filter: env-filter.bash
+	cp $< $@
+	chmod +x $@
 
 .PHONY: all
 all: $(ALL)

--- a/package-builder/build-manifest.nix
+++ b/package-builder/build-manifest.nix
@@ -12,6 +12,8 @@
   buildDeps ? [ ], # optional
   buildScript ? null, # optional
   buildCache ? null, # optional
+  allowEnvVars ? [ ], # variables to allow into build from outer dev env
+  allowEnvVarPrefixes ? [ ], # prefixes of variables allowed into build
 }:
 # First a few assertions to ensure that the inputs are consistent.
 # buildCache is only meaningful with a build script
@@ -46,6 +48,10 @@ let
     ${dollar_out_bin_copy_hints}
   '';
   name = "${pname}-${version}";
+  envFilterAllowArgs = builtins.concatStringsSep " " (
+    builtins.map (x: "--allow ${x}") allowEnvVars
+    ++ builtins.map (x: "--allow-prefix ${x}") allowEnvVarPrefixes
+  );
 in
 pkgs.runCommandNoCC name
   {
@@ -166,9 +172,10 @@ pkgs.runCommandNoCC name
                 # N.B. not using t3 --forcecolor option because Nix sandbox
                 # strips color codes from output anyway.
                 FLOX_SRC_DIR=$(pwd) FLOX_RUNTIME_DIR="$TMP" \
-                  ${flox-env-package}/activate --env ${flox-env-package} --env-project $(pwd) --mode build-sandbox -- \
-                    ${build-wrapper-env-package}/wrapper --env ${build-wrapper-env-package} --set-vars -- \
-                      t3 --relative $log -- bash -e ${buildScript-contents}
+                  ${flox-env-package}/activate --env ${flox-env-package} --mode build --env-project $(pwd) -- \
+                    @envFilter@ ${envFilterAllowArgs} -- \
+                      ${build-wrapper-env-package}/wrapper --env ${build-wrapper-env-package} --set-vars -- \
+                        t3 --relative $log -- bash -e ${buildScript-contents}
               ''
             else
               ''
@@ -180,25 +187,13 @@ pkgs.runCommandNoCC name
                 # TMP will be set to something like
                 # /private/tmp/nix-build-file-0.0.0.drv-0
 
-                # See flox-build.mk for more explanation of why we use a nested
-                # activation.
-                # Just like for impure builds, we want to prefer the build
-                # wrapper environment for tools and libraries.
-                # Unlike for impure builds, we use run mode since we're leaning
-                # on Nix to setup languages.
-                # This may be happening because flox-env-package is in
-                # buildInputs,
-                # but we would need to confirm.
-                # Because we use run mode instead of dev mode, we don't need to
-                # filter out anything set by dev mode, so we don't use env -i
-                # Strictly speaking we don't need --set-vars, since the outer
-                # activation will set those,
-                # and we're not using env -i like we are in the impure build
-
+                # See flox-build.mk for a detailed explanation of why we use a nested
+                # activation when performing builds.
                 FLOX_SRC_DIR=$(pwd) FLOX_RUNTIME_DIR="$TMP" \
-                  ${flox-env-package}/activate --env ${flox-env-package} --env-project $(pwd) --mode build-sandbox -- \
-                    ${build-wrapper-env-package}/wrapper --env ${build-wrapper-env-package} --set-vars -- \
-                      t3 --relative $log -- bash -e ${buildScript-contents} || \
+                  ${flox-env-package}/activate --env ${flox-env-package} --mode build --env-project $(pwd) -- \
+                    @envFilter@ ${envFilterAllowArgs} -- \
+                      ${build-wrapper-env-package}/wrapper --env ${build-wrapper-env-package} --set-vars -- \
+                        t3 --relative $log -- bash -e ${buildScript-contents} || \
                 ( rm -rf $out && echo "flox build failed (caching build dir)" | tee $out 1>&2 )
               ''
           }

--- a/package-builder/build-manifest.nix
+++ b/package-builder/build-manifest.nix
@@ -12,6 +12,7 @@
   buildDeps ? [ ], # optional
   buildScript ? null, # optional
   buildCache ? null, # optional
+  packageBuilder ? "@out@", # path to find adjacent env-filter script
   allowEnvVars ? [ ], # variables to allow into build from outer dev env
   allowEnvVarPrefixes ? [ ], # prefixes of variables allowed into build
 }:
@@ -23,7 +24,11 @@ assert (srcTarball != null) -> (buildScript != null);
 let
   flox-env-package = builtins.storePath flox-env;
   build-wrapper-env-package = builtins.storePath build-wrapper-env;
+  # We need a reference to the package that this file comes from so that
+  # we can pull the adjacent env-filter script into the build.
+  package-builder-package = builtins.storePath packageBuilder;
   buildInputs = [
+    package-builder-package
     build-wrapper-env-package
     flox-env-package
   ] ++ (map (d: builtins.storePath d) buildDeps);
@@ -173,7 +178,7 @@ pkgs.runCommandNoCC name
                 # strips color codes from output anyway.
                 FLOX_SRC_DIR=$(pwd) FLOX_RUNTIME_DIR="$TMP" \
                   ${flox-env-package}/activate --env ${flox-env-package} --mode build --env-project $(pwd) -- \
-                    @envFilter@ ${envFilterAllowArgs} -- \
+                    ${package-builder-package}/libexec/env-filter ${envFilterAllowArgs} -- \
                       ${build-wrapper-env-package}/wrapper --env ${build-wrapper-env-package} --set-vars -- \
                         t3 --relative $log -- bash -e ${buildScript-contents}
               ''
@@ -191,7 +196,7 @@ pkgs.runCommandNoCC name
                 # activation when performing builds.
                 FLOX_SRC_DIR=$(pwd) FLOX_RUNTIME_DIR="$TMP" \
                   ${flox-env-package}/activate --env ${flox-env-package} --mode build --env-project $(pwd) -- \
-                    @envFilter@ ${envFilterAllowArgs} -- \
+                    ${package-builder-package}/libexec/env-filter ${envFilterAllowArgs} -- \
                       ${build-wrapper-env-package}/wrapper --env ${build-wrapper-env-package} --set-vars -- \
                         t3 --relative $log -- bash -e ${buildScript-contents} || \
                 ( rm -rf $out && echo "flox build failed (caching build dir)" | tee $out 1>&2 )

--- a/package-builder/env-filter.bash
+++ b/package-builder/env-filter.bash
@@ -1,3 +1,4 @@
+#!/bin/bash
 
 # env-filter: redact environment variables, apart from those expressly called out
 # with (-a|--allow) args or matching prefixes provided with (-p|--allow-prefix).

--- a/package-builder/flox-build.mk
+++ b/package-builder/flox-build.mk
@@ -51,7 +51,6 @@ _cat := $(call __package_bin,$(__coreutils),cat)
 _cp := $(call __package_bin,$(__coreutils),cp)
 _cut := $(call __package_bin,$(__coreutils),cut)
 _daemonize := $(call __package_bin,$(__daemonize),daemonize)
-_envFilter := @envFilter@
 _git := $(call __package_bin,$(__gitMinimal),git)
 _grep := $(call __package_bin,$(__gnugrep),grep)
 _head := $(call __package_bin,$(__coreutils),head)
@@ -349,12 +348,12 @@ ALLOW_OUTER_ENV_PREFIXES = _
 # Add vars set by Nix stdenv hooks.
 ALLOW_OUTER_ENV_PREFIXES += NIX_CFLAGS NIX_CC
 
-# Assemble list of all $(_envFilter) "allow" args. First start by
+# Assemble list of all env-filter "allow" args. First start by
 # quoting the variable names so that they can be passed in the shell
 # and as arguments to build-manifest.nix.
 QUOTED_ALLOW_OUTER_ENV_VARS = $(foreach _v,$(ALLOW_OUTER_ENV_VARS),"$(_v)")
 QUOTED_ALLOW_OUTER_ENV_VAR_PREFIXES = $(foreach _p,$(ALLOW_OUTER_ENV_PREFIXES),"$(_p)")
-envFilter_ALLOW_ARGS = \
+env_filter_ALLOW_ARGS = \
   $(foreach _v,$(QUOTED_ALLOW_OUTER_ENV_VARS),--allow $(_v)) \
   $(foreach _p,$(QUOTED_ALLOW_OUTER_ENV_VAR_PREFIXES),--allow-prefix $(_p))
 
@@ -392,13 +391,13 @@ define BUILD_local_template =
   # We could use `env -i` to prevent all variables from leaking from the
   # "develop" environment into the inner activation, but then that causes
   # problems for compilers that rely on NIX_CC* environment variables set in
-  # the outer activation. To address this problem we use the `envFilter` script
+  # the outer activation. To address this problem we use the `env-filter` script
   # to remove all variables not specifically allowed through to the inner
   # activation.
 
   # The final result is approximately the following:
   #   $(FLOX_INTERPRETER)/activate ... -- \
-  #     $(_envFilter) $(envFilter_ALLOW_ARGS) -- \
+  #     $(_libexec_dir)/env-filter $(env_filter_ALLOW_ARGS) -- \
   #       $(_build_wrapper_env)/wrapper ... -- bash -e buildScript
   $($(_pvarname)_out) $($(_pvarname)_logfile): $($(_pvarname)_buildScript)
 	@# $(if $(FLOX_INTERPRETER),,$$(error FLOX_INTERPRETER not defined))
@@ -407,7 +406,7 @@ define BUILD_local_template =
 	$(_V_) out=$($(_pvarname)_out) \
 	  $(if $(_virtualSandbox),$(PRELOAD_VARS) FLOX_SRC_DIR=$$$$($(_pwd)) FLOX_VIRTUAL_SANDBOX=$(_sandbox)) \
 	  $(FLOX_INTERPRETER)/activate --env $(FLOX_ENV) --mode build --env-project $$$$($(_pwd)) -- \
-	    $(_envFilter) $(envFilter_ALLOW_ARGS) -- \
+	    $(_libexec_dir)/env-filter $(env_filter_ALLOW_ARGS) -- \
 	      $(_build_wrapper_env)/wrapper --env $(_build_wrapper_env) --set-vars -- \
 	        $(_t3) $($(_pvarname)_logfile) -- $(_bash) -e $$<
 

--- a/package-builder/flox-build.mk
+++ b/package-builder/flox-build.mk
@@ -70,6 +70,7 @@ _sed := $(call __package_bin,$(__gnused),sed)
 _sha256sum := $(call __package_bin,$(__coreutils),sha256sum)
 _tar := $(call __package_bin,$(__gnutar),tar)
 _t3 := $(call __package_bin,$(__t3),t3) --relative $(if $(NO_COLOR),,--forcecolor)
+_tr := $(call __package_bin,$(__coreutils),tr)
 _uname := $(call __package_bin,$(__coreutils),uname)
 
 # Identify path to build-manifest.nix, in same directory as this Makefile.
@@ -611,10 +612,13 @@ define MANIFEST_BUILD_template =
 	$$(eval _build_closure_extra_packages = $$(strip \
 	  $$(filter-out $$(_build_store_path) $$(_build_wrapper_requisites) $$(_nef_requisites), \
 	    $$(_build_closure_requisites))))
+	$$(eval _count = $$(words $$(_build_closure_extra_packages)))
+	$$(eval _space = $$(shell echo $$(_count) | $(_tr) '[0-9]' '-'))
 	$$(if $$(_build_closure_extra_packages),$(_VV_) \
-	  echo -e "❌ packages found in $$(_build_store_path)\n" \
-	           "       not found in $(_build_wrapper_env)\n" 1>&2; \
-	  $$(foreach _pkg,$$(_build_closure_extra_packages), \
+	  echo -e "❌ $$(_count) packages found in $$(_build_store_path)\n" \
+	           "  $$(_space)      not found in $(_build_wrapper_env)\n" 1>&2; \
+	  $$(intcmp 3,$$(_count),echo -e "Displaying first 3 only:\n" 1>&2; ) \
+	  $$(foreach _pkg,$$(wordlist 1,3,$$(_build_closure_extra_packages)), \
 	    ( $(_nix) why-depends --precise $$(_build_store_path) $$(_pkg) && echo ) 1>&2; ) \
 	  exit 1)
 	@# TODO: Strip the buildCache and log outputs of all requisites.

--- a/package-builder/flox-build.mk
+++ b/package-builder/flox-build.mk
@@ -51,7 +51,6 @@ _cat := $(call __package_bin,$(__coreutils),cat)
 _cp := $(call __package_bin,$(__coreutils),cp)
 _cut := $(call __package_bin,$(__coreutils),cut)
 _daemonize := $(call __package_bin,$(__daemonize),daemonize)
-_env := $(call __package_bin,$(__coreutils),env)
 _envFilter := @envFilter@
 _git := $(call __package_bin,$(__gitMinimal),git)
 _grep := $(call __package_bin,$(__gnugrep),grep)
@@ -324,8 +323,10 @@ endef
 # of the tools and compilers found in the outer environment, and for Flox to
 # otherwise function properly.
 #
+# Start with output path created by the build.
+ALLOW_OUTER_ENV_VARS = out
 # Vars required for flox functionality.
-ALLOW_OUTER_ENV_VARS := \
+ALLOW_OUTER_ENV_VARS += \
   FLOX_ACTIVATE_TRACE FLOX_ENV_DIRS FLOX_RUNTIME_DIR HOME PATH
 # Vars exported by 0100_common-paths.sh. Not included:
 # - LIBRARY_PATH : embedded in runpath of executables
@@ -340,8 +341,22 @@ ALLOW_OUTER_ENV_VARS += RUST_SRC_PATH
 ALLOW_OUTER_ENV_VARS += JUPYTER_PATH
 # Vars exported by 0800_cuda.sh.
 ALLOW_OUTER_ENV_VARS += LD_FLOXLIB_FILES_PATH
-# Vars set by Nix stdenv hooks.
-ALLOW_OUTER_ENV_VARS += $(filter NIX_CFLAGS% NIX_CC%,$(.VARIABLES))
+
+# Env var prefixes that are allowed to be passed from the outer
+# "develop" environment to the inner "build wrapper" environment.
+# Start by allowing any variable starting with "_".
+ALLOW_OUTER_ENV_PREFIXES = _
+# Add vars set by Nix stdenv hooks.
+ALLOW_OUTER_ENV_PREFIXES += NIX_CFLAGS NIX_CC
+
+# Assemble list of all $(_envFilter) "allow" args. First start by
+# quoting the variable names so that they can be passed in the shell
+# and as arguments to build-manifest.nix.
+QUOTED_ALLOW_OUTER_ENV_VARS = $(foreach _v,$(ALLOW_OUTER_ENV_VARS),"$(_v)")
+QUOTED_ALLOW_OUTER_ENV_VAR_PREFIXES = $(foreach _p,$(ALLOW_OUTER_ENV_PREFIXES),"$(_p)")
+envFilter_ALLOW_ARGS = \
+  $(foreach _v,$(QUOTED_ALLOW_OUTER_ENV_VARS),--allow $(_v)) \
+  $(foreach _p,$(QUOTED_ALLOW_OUTER_ENV_VAR_PREFIXES),--allow-prefix $(_p))
 
 # The following template renders targets for the in-situ build mode.
 define BUILD_local_template =
@@ -374,33 +389,32 @@ define BUILD_local_template =
   # activations can prepend to a PYTHONPATH that gets embedded in the
   # build, which has the effect of pulling both environments into the closure.
 
-  # We can use `env -i` to prevent that leakage of the "develop" environment
-  # path into the inner activation, but then that causes problems for compilers
-  # that rely on NIX_CC* environment variables set in the outer activation. To
-  # address this problem we maintain a list of ALLOW_OUTER_ENV_VARS allowed
-  # to be propagated from the outer to the inner activation, and again use the
-  # `env` command to let those through.
+  # We could use `env -i` to prevent all variables from leaking from the
+  # "develop" environment into the inner activation, but then that causes
+  # problems for compilers that rely on NIX_CC* environment variables set in
+  # the outer activation. To address this problem we use the `envFilter` script
+  # to remove all variables not specifically allowed through to the inner
+  # activation.
 
   # The final result is approximately the following:
   #   $(FLOX_INTERPRETER)/activate ... -- \
-  #     $(_envFilter) --allow-prefix _ $(foreach v,$(ALLOW_OUTER_ENV_VARS),--allow $(v)) -- \
+  #     $(_envFilter) $(envFilter_ALLOW_ARGS) -- \
   #       $(_build_wrapper_env)/wrapper ... -- bash -e buildScript
   $($(_pvarname)_out) $($(_pvarname)_logfile): $($(_pvarname)_buildScript)
 	@# $(if $(FLOX_INTERPRETER),,$$(error FLOX_INTERPRETER not defined))
 	@echo "Building $(_name) in local mode"
 	$(_VV_) $(_rm) -rf $($(_pvarname)_out)
-	$(_V_) \
+	$(_V_) out=$($(_pvarname)_out) \
 	  $(if $(_virtualSandbox),$(PRELOAD_VARS) FLOX_SRC_DIR=$$$$($(_pwd)) FLOX_VIRTUAL_SANDBOX=$(_sandbox)) \
 	  $(FLOX_INTERPRETER)/activate --env $(FLOX_ENV) --mode build --env-project $$$$($(_pwd)) -- \
-	    $(_envFilter) --allow-prefix _ $(foreach v,$(ALLOW_OUTER_ENV_VARS),--allow $(v)) -- \
+	    $(_envFilter) $(envFilter_ALLOW_ARGS) -- \
 	      $(_build_wrapper_env)/wrapper --env $(_build_wrapper_env) --set-vars -- \
-	        $(_t3) $($(_pvarname)_logfile) -- $(_env) out=$($(_pvarname)_out) $(_bash) -e $$<
+	        $(_t3) $($(_pvarname)_logfile) -- $(_bash) -e $$<
 
   # Having built the package to $($(_pvarname)_out) outside of Nix, call
   # build-manifest.nix to turn it into a Nix package.
   $($(_pvarname)_buildJSON): $($(_pvarname)_out)
-	$(_V_) set -o pipefail && \
-	$(_nix) build -L --file $(_libexec_dir)/build-manifest.nix \
+	$(_V_) $(_nix) build -L --file $(_libexec_dir)/build-manifest.nix \
 	  --argstr pname "$(_pname)" \
 	  --argstr version "$(_version)" \
 	  --argstr flox-env "$(FLOX_ENV)" \
@@ -408,6 +422,8 @@ define BUILD_local_template =
 	  --argstr install-prefix "$($(_pvarname)_out)" \
 	  --argstr nixpkgs-url "$(BUILDTIME_NIXPKGS_URL)" \
 	  $(if $($(_pvarname)_buildDeps),--arg buildDeps '[$($(_pvarname)_buildDeps)]') \
+	  --arg allowEnvVars '[$(QUOTED_ALLOW_OUTER_ENV_VARS)]' \
+	  --arg allowEnvVarPrefixes '[$(QUOTED_ALLOW_OUTER_ENV_VAR_PREFIXES)]' \
 	  --out-link "result-$(_pname)" \
 	  --json '^*' > $$@
 
@@ -484,6 +500,8 @@ define BUILD_nix_sandbox_template =
 	  --argstr build-wrapper-env "$(_build_wrapper_env)" \
 	  --argstr nixpkgs-url "$(BUILDTIME_NIXPKGS_URL)" \
 	  $(if $($(_pvarname)_buildDeps),--arg buildDeps '[$($(_pvarname)_buildDeps)]') \
+	  --arg allowEnvVars '[$(QUOTED_ALLOW_OUTER_ENV_VARS)]' \
+	  --arg allowEnvVarPrefixes '[$(QUOTED_ALLOW_OUTER_ENV_VAR_PREFIXES)]' \
 	  --argstr buildScript "$($(_pvarname)_buildScript)" \
 	  $(if $(_do_buildCache),--argstr buildCache "$($(_pvarname)_buildCache)") \
 	  --out-link "result-$(_pname)" \

--- a/pkgs/flox-package-builder/default.nix
+++ b/pkgs/flox-package-builder/default.nix
@@ -10,42 +10,8 @@
   jq,
   nix,
   stdenv,
-  substituteAll,
   t3,
-  writeShellScript,
 }:
-let
-  envFilter = writeShellScript "env-filter" (
-    builtins.readFile (substituteAll {
-      src = ../../package-builder/env-filter.bash;
-      env = {
-        inherit coreutils getopt;
-      };
-    })
-  );
-  build-manifest-nix = substituteAll {
-    name = "build-manifest.nix";
-    src = ../../package-builder/build-manifest.nix;
-    inherit envFilter;
-  };
-  flox-build-mk = substituteAll {
-    name = "flox-build.mk";
-    src = ../../package-builder/flox-build.mk;
-    inherit
-      bashInteractive
-      coreutils
-      daemonize
-      envFilter
-      gitMinimal
-      gnugrep
-      gnused
-      gnutar
-      jq
-      nix
-      t3
-      ;
-  };
-in
 stdenv.mkDerivation {
   pname = "package-builder";
   version = "1.0.0";
@@ -54,8 +20,20 @@ stdenv.mkDerivation {
     path = "${./../../package-builder}";
   };
   postPatch = ''
-    cp ${flox-build-mk} flox-build.mk
-    cp ${build-manifest-nix} build-manifest.nix
+    # Need to perform substitutions within derivation for access to $out.
+    for i in build-manifest.nix env-filter.bash flox-build.mk; do
+      bashInteractive=${bashInteractive} \
+      coreutils=${coreutils} \
+      daemonize=${daemonize} \
+      getopt=${getopt} \
+      gitMinimal=${gitMinimal} \
+      gnugrep=${gnugrep} \
+      gnused=${gnused} \
+      gnutar=${gnutar} \
+      jq=${jq} \
+      nix=${nix} \
+      t3=${t3} substituteAllInPlace $i
+    done
   '';
   # install the packages to $out/libexec/*
   makeFlags = [ "PREFIX=$(out)" ];

--- a/pkgs/flox-package-builder/default.nix
+++ b/pkgs/flox-package-builder/default.nix
@@ -26,6 +26,7 @@ let
   build-manifest-nix = substituteAll {
     name = "build-manifest.nix";
     src = ../../package-builder/build-manifest.nix;
+    inherit envFilter;
   };
   flox-build-mk = substituteAll {
     name = "flox-build.mk";

